include/gui/application.hpp

```diff
@@ -180,23 +179,38 @@ namespace Toolbox {
 
     class FileDialog {
     public:
-        static FileDialog* Instance() {
+        FileDialog() = default;
+        ~FileDialog() {
+            if (m_thread_initialized) {
+                m_thread.join();
+            }
+        }
+
+        static FileDialog *Instance() {
             static FileDialog _instance;
             return &_instance;
         }
-        void OpenDialog(std::filesystem::path starting_path,
-                        GLFWwindow* parent_window,
+        void OpenDialog(std::filesystem::path starting_path, GLFWwindow *parent_window,
                         bool is_directory = false,
-                        std::optional<std::vector<std::pair<
-                        const char*, const char*>>> maybe_filters = std::nullopt);
+                        std::optional<std::vector<std::pair<std::string, std::string>>>
+                            maybe_filters = std::nullopt);
         bool isAlreadyOpen() { return m_thread_running; }
-        bool isDone() { return !m_thread_running && not m_closed; }
+        bool isDone() { return !m_thread_running && !m_closed && m_thread_initialized; }
```

Comment:
Oops yeah I've been writing too much python, I'm surprised the compiler accepts "not" now. Good call on making sure the thread is initialized.